### DISCUSSION TITLE
🖋Markdown神啊!(整理與分享)

### DISCUSSION THREAD

**Discussion by JunTingLin** [1101-GS4719-C/Python-Community-GS4719-C#377] [General]
# :loudspeaker:``分享Markdown語法``

## 前言
github上面會看到md檔，md檔裡面其實撰寫的是Markdown的語法，透過語法可以實現用語法控制粗體、加icon🖥、加底線、刪除現等等，因為我本身也不熟悉，所以想藉此機會上網查詢常用的幾種Markdown語法，整理並分享給各位🎉。

## 甚麼是Markdown?
markdown語言是屬於標記語言。HTML與Markdown，兩者是相輔相成，HTML就像是稿紙上的格線，規劃整體的格式；而Markdown就是裡面的標點符號，讓文件容易閱讀及編寫。因此，Markdown的語法全由標點符號組成。「易讀易寫」就是Markdown的精神💪。

## 我哪裡會需要下MarkDown
剛剛有提過github上面的md檔都是要用Markdown語法來撰寫的，才可以達到票漂亮亮的文字呈現，另外分享一款用Markdown作筆記的軟體:[hackmd](https://hackmd.io/ "hackmd")

![GitHub](https://i.imgur.com/HMXa5QS.gif)

用它來做筆記超讚的，聽說有許多套件可以用，就算複雜的數學符號也ok，而且還可以多人共享共筆。

### 標題(H1~H6)
```
# This is an H1
## This is an H2
###### This is an H6
```
效果就如我上面的那些標題⏫

### 行分隔符
```
***
_______
-------------------
```
效果就如下面的三條線:
***
_______
-------------------

#### 無序
```
* 星號
+ 加號
- 減號
```
效果如下:
* 星號
+ 加號
- 減號

#### 有序
```
1. 有序
2. 清單
```
*記得:"1."完後面記的要在空一格*
1. 有序
2. 清單

#### 普通連結
`[Google](http://www.google.com/)`

[Google](http://www.google.com/)

#### 指向本地文件的鏈接
`[icon.png](./images/icon.png)`

這種寫法可以指向本地儲存庫的連結位置

### 插入圖片
`![GitHub](https://imgur.dcard.tw/Y1ecdMy.gif)`

![GitHub](https://imgur.dcard.tw/Y1ecdMy.gif)

### 表格
```
|    name    | age |
| ---------- | --- |
| LearnShare |  12 |
| Mike       |  32 |
```
使用 |來區分不同的單元格，使用 - 來分隔表頭和其他行
|    name    | age |
| ---------- | --- |
| LearnShare |  12 |
| Mike       |  32 |

## 結語
這些應該是比較常用到的Markdown語法，藉由分享也順便學習與熟練，假如還有需要歡迎補充😁

## 參考資料
+ [3分鐘學會Markdown常用語法](https://tiida54.github.io/2018/01/03/3%E5%88%86%E9%90%98%E5%AD%B8%E6%9C%83Markdown%E5%B8%B8%E7%94%A8%E8%AA%9E%E6%B3%95/)

+ [Markdown語法簡介](https://xianbai.me/learn-md/article/syntax/links.html)

+ 以上圖片均在google搜尋

**Comment by handsomeniko** [comment]
挖這個好好用，長知識了~~~

**Comment by pekoranoinu** [comment]
謝謝分享Markdown的使用

**Comment by chupochun** [comment]
謝謝分享 我又多了一種作筆記的方法了

**Comment by xkllkx** [comment]
謝謝分享，剛好對Markdown的語法不熟

**Comment by lx3211** [comment]
推推hackmd好用

**Comment by ThomasMeng0515** [comment]
感謝分享，讓我對md語法有初步認識~

**Comment by ghost** [comment]
以後md語法不懂先來看這篇ww，然後卡波可愛><

**Comment by lx3211** [comment]
咖波照片的網址是怎麼獲得的啊？

**Reply by JunTingLin** [reply]
到網路上搜尋想要的圖片->選擇圖片按右鍵->複製圖片->貼上即可

![image](https://truth.bahamut.com.tw/s01/201610/e3f1da59c3f886dec6c4e74be89670e8.GIF)

**Reply by JunTingLin** [reply]
![螢幕擷取畫面 2022-03-03 133406](https://user-images.githubusercontent.com/92431095/156503797-2c59c8ad-6ca6-4adb-84de-4e2bda1413c5.png)

**Reply by JunTingLin** [reply]
不好意思這邊更正並補充一下😅
有分兩種
1. 假如貼的是靜態圖片
直接右鍵->**複製圖片**->貼上這~~即可

2. 可是假如要是會動的gif圖檔
必須使用markdown語法，如下
`![image](https://truth.bahamut.com.tw/s01/201610/e3f1da59c3f886dec6c4e74be89670e8.GIF)`
此時網路圖片那裡要改點選 右鍵->**複製圖片位置**

**Reply by lx3211** [reply]
謝謝回答～

**Comment by sean1325418** [comment]
想發個新文章剛好可以現學現用

**Comment by cordydeng** [comment]
太棒了，最近剛好在學html，不過不知道要把重點寫在哪裡。好在有這篇文章，讓我知道了一個在電腦上做筆記的方法

**Reply by JunTingLin** [reply]
Markdown 最終還是要轉換成 HTML，所以我想假如在markdown語法中直接使用html語法，比如換行<br>應該也是可行的

**Comment by JSHT** [comment]
要使用 hackmd 做筆記的話，這邊還可以搭配 [Hackmd 完整教學](https://hackmd.io/c/tutorials-tw/%2Fs%2Ftutorials-tw) 了解各種神奇功能。

**Comment by Yuhsin000** [comment]
謝分享，capoo 超可愛

**Comment by JunTingLin** [comment]
上面剛好有同學提到HTML，這邊來補充一下
圖片可以直接複製圖片並貼上於此
![image](https://user-images.githubusercontent.com/92431095/157410537-67ce6632-6ad0-42f7-8592-f28cadce2668.png)
可是爆大的(超過1000px X 1000px)
此時就可以改用html標籤
```
<img src="https://user-images.githubusercontent.com/92431095/157406561-bae21182-a2f5-4fa1-a4b6-be0849dad18d.png" width = "200" height = "200" alt="NCU logo" />
```
<img src="https://user-images.githubusercontent.com/92431095/157406561-bae21182-a2f5-4fa1-a4b6-be0849dad18d.png" width = "200" height = "200" alt="NCU logo" />
484就可以下參數改變大小了😁

**Reply by chia-kai-chang** [reply]
確實markdown無法調整圖片大小，只能用html的`img`**標籤**的`width`與`height`**屬性**來調整了~

**Comment by henrywen547** [comment]
有這篇Markdown教學真的很方便!有提到了大部分時候會使用的語法，這樣就不用每次都去看github的全英文說明了!
想知道更多使用方法也可以前往Github Docs!
[Github Docs](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax)

**Comment by ThomasMeng0515** [comment]
剛剛在撰寫文章的時候發現，也可以直接把圖片用拖曳的拖到文章裡面，同樣能成功顯示出圖片，給大家參考!

**Comment by JunTingLin** [comment]
本篇上面提到的強大Hackmd
我已經補充在本篇[📚以後作筆記就靠HackMD啦!數學公式想寫的漂漂亮亮、想做PPT他全都罩 #516](https://github.com/NCU-GS4719-Python/Python-Community/discussions/516)囉!!

**Comment by jason79461385** [comment]
也可以透過github自動產出gif檔案後，再寫到markdown的文件中，我主頁貪吃蛇就是這樣用的。

**Comment by minniewei** [comment]
感謝大大方分享，最近一直聽到這東東但卻不知道怎麼用！哭

**Comment by treeann** [comment]
感謝分享 超級實用

**Comment by jcpan821** [comment]
感謝分享~~

**Comment by alexthebigpp** [comment]
感謝分享，又對github有進一步的瞭解
